src/stdio.rs

```diff
@@ -128,6 +128,20 @@ impl_try_from_tokio_process_child_for_stdio!(ChildStderr);
 #[derive(Debug)]
 pub struct ChildStdin(tokio_pipe::PipeWrite);
 
+#[cfg(feature = "openssh-sftp-client")]
+impl openssh_sftp_client::Writer for ChildStdin {
```

Comment:
Can we move this `impl` to `openssh_sftp_client`?

Reply:
I am afraid we can't.

The efficient implementation of this trait requires access to the underlying PipeWrite type.

Reply:
One way to solve this is to add a new method into_inner, which directly exposes the underlying PipeWrite, but this again, exposes implementation details.

Reply:
Perhaps I can move this trait into another crate, such as tokio-io-utility,

Reply:
That would work. I'm still surprised that this extra trait is even necessary. It feels like a big ergonomic hit for what seems like questionable gain over just using, say, `AsyncWrite::poll_write_vectored`.

Reply:
I could remove this requirement by removing the atomic write support in `WriteEnd`.

Perhaps I really should think of an alternative design for `openssh_sftp_client` to make its internal more simpler.

Reply:
I think that might be valuable. In general it feels excessive for the protocol to require that consumers implement as low-level a trait as `AsyncWrite` themselves in a custom way.

Reply:
> I think that might be valuable. In general it feels excessive for the protocol to require that consumers implement as low-level a trait as `AsyncWrite` themselves in a custom way.

Yeah, it is a bit excessive, though most users probably wouldn't notice as they just pass PipeWrite to it.

Reply:
@jonhoo It's done!

[openssh-sftp-client-0.10.0](https://docs.rs/openssh-sftp-client/0.10.0/openssh_sftp_client/) is released with the refactor done, so we no longer has to depend on `openssh-sftp-client`.

Reply:
Excellent, thank you!

And totally understandable that things may take longer when you're busy — I've been a prime example of that :)